x-pack/plugins/data_visualizer/public/application/common/components/top_values/top_values.tsx

```diff
@@ -47,75 +48,120 @@ function getPercentLabel(percent: number): string {
-          id="xpack.dataVisualizer.dataGrid.field.topValues.calculatedFromTotalRecordsLabel"
-          defaultMessage="Calculated from {totalDocumentsFormatted} {totalDocuments, plural, one {record} other {records}}."
-          values={{
-            totalDocuments,
-            totalDocumentsFormatted: (
-              <strong>
-                {fieldFormats
-                  .getDefaultInstance(KBN_FIELD_TYPES.NUMBER, [ES_FIELD_TYPES.INTEGER])
-                  .convert(totalDocuments ?? 0)}
+                  .convert(stats.topValuesSampleSize)}
               </strong>
             ),
           }}
         />
-      )}
+      );
+    }
+
+    return totalDocuments > (sampleCount ?? 0) ? (
+      <FormattedMessage
+        id="xpack.dataVisualizer.dataGrid.field.topValues.calculatedFromSampleRecordsLabel"
+        defaultMessage="Calculated from {sampledDocumentsFormatted} sample {sampledDocuments, plural, one {record} other {records}}."
+        values={{
+          sampledDocuments: sampleCount,
+          sampledDocumentsFormatted: (
+            <strong>
+              {fieldFormats
+                .getDefaultInstance(KBN_FIELD_TYPES.NUMBER, [ES_FIELD_TYPES.INTEGER])
+                .convert(sampleCount)}
+            </strong>
+          ),
+        }}
+      />
+    ) : (
+      <FormattedMessage
+        id="xpack.dataVisualizer.dataGrid.field.topValues.calculatedFromTotalRecordsLabel"
+        defaultMessage="Calculated from {totalDocumentsFormatted} {totalDocuments, plural, one {record} other {records}}."
+        values={{
+          totalDocuments,
+          totalDocumentsFormatted: (
+            <strong>
+              {fieldFormats
+                .getDefaultInstance(KBN_FIELD_TYPES.NUMBER, [ES_FIELD_TYPES.INTEGER])
+                .convert(totalDocuments ?? 0)}
+            </strong>
+          ),
+        }}
+      />
+    );
+  };
+  const countsElement = (
+    <EuiText color="subdued" size="xs">
+      {getMessage()}
     </EuiText>
   );
 
+  const topValues = originalTopValues.map((bucket) => ({
+    ...bucket,
+    percent:
+      typeof bucket.percent === 'number' ? bucket.percent : bucket.doc_count / totalDocuments,
+  }));
+
+  const topValuesOtherCountPercent =
+    1 - (topValues ? topValues.reduce((acc, bucket) => acc + bucket.percent, 0) : 0);
+  const topValuesOtherCount = Math.floor(topValuesOtherCountPercent * (sampleCount ?? 0));
+
   return (
     <ExpandedRowPanel
       dataTestSubj={'dataVisualizerFieldDataTopValues'}
       className={classNames('dvPanel__wrapper', compressed ? 'dvPanel--compressed' : undefined)}
     >
       <ExpandedRowFieldHeader>
-        <FormattedMessage
-          id="xpack.dataVisualizer.dataGrid.field.topValuesLabel"
-          defaultMessage="Top values"
-        />
+        {showSampledValues ? (
+          <FormattedMessage
+            id="xpack.dataVisualizer.dataGrid.field.topSampledValuesLabel"
+            defaultMessage="Top Sampled values"
```

Comment:
Nit (as it's being converted to uppercase) but `Top sampled values` for sentence case.